test/plugin/example/example_plugin.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bufio"
+	"net"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+type ExamplePlugin struct{}
+
+// connectorFunc is the function that implements the tcp.Connector func signature in this
+// example plugin. It has access to the client connection and the secrets (as a map),
+// and is expected to return the target service connection.
+//
+// This example connector works as follows:
+// 1. Waits for the initial message from the client
+// 2. Connect to a target service whose address is the value of the secret identified by
+// the key "address"
+// 3. Inject credentials from a secret identified by the key "auth"
+// 4. Write the initial message from the client with some modification
+func connectorFunc(clientConn net.Conn, secrets plugin.SecretsByID) (net.Conn, error) {
+	r := bufio.NewReader(clientConn)
+	res, _, err := r.ReadLine()
```

Comment:
Because the distance from this declaration to its usage below is 14 lines, I would use the name `clientResponse` or something similar.  `r` is fine for reader, otoh, because the span of its usage is exactly 1 line away from the declaration.